pkg/utils/kube/ctl.go

```diff
@@ -328,6 +318,80 @@ func kubeCmdFactory(kubeconfig, ns string) (cmdutil.Factory, genericclioptions.I
 	return f, ioStreams
 }
 
+func (k *KubectlCmd) ReplaceResource(ctx context.Context, config *rest.Config, obj *unstructured.Unstructured, namespace string, dryRunStrategy cmdutil.DryRunStrategy, force bool) (string, error) {
+	span := k.Tracer.StartSpan("ReplaceResource")
+	span.SetBaggageItem("kind", obj.GetKind())
+	span.SetBaggageItem("name", obj.GetName())
+	defer span.Finish()
+	k.Log.Info(fmt.Sprintf("Replacing resource %s/%s in cluster: %s, namespace: %s", obj.GetKind(), obj.GetName(), config.Host, namespace))
+	return k.runResourceCommand(ctx, config, obj, namespace, dryRunStrategy, func(config *rest.Config, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, fileName string, namespace string, dryRunStrategy cmdutil.DryRunStrategy) error {
+		cleanup, err := k.processKubectlRun("replace")
+		if err != nil {
+			return err
+		}
+		defer cleanup()
+
+		replaceOptions, err := newReplaceOptions(config, f, ioStreams, fileName, namespace, force, dryRunStrategy)
+		if err != nil {
+			return err
+		}
+		return replaceOptions.Run(f)
+	})
+}
+
+func (k *KubectlCmd) CreateResource(ctx context.Context, config *rest.Config, obj *unstructured.Unstructured, namespace string, dryRunStrategy cmdutil.DryRunStrategy) (*unstructured.Unstructured, error) {
+	gvk := obj.GroupVersionKind()
+	span := k.Tracer.StartSpan("CreateResource")
+	span.SetBaggageItem("kind", gvk.Kind)
+	span.SetBaggageItem("name", obj.GetName())
+	defer span.Finish()
+	dynamicIf, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	disco, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	apiResource, err := ServerResourceForGroupVersionKind(disco, gvk)
+	if err != nil {
+		return nil, err
+	}
+	resource := gvk.GroupVersion().WithResource(apiResource.Name)
+	resourceIf := ToResourceInterface(dynamicIf, apiResource, resource, namespace)
+
+	createOptions := metav1.CreateOptions{}
+	switch dryRunStrategy {
+	case cmdutil.DryRunClient:
+		createOptions.DryRun = []string{metav1.DryRunAll}
+	case cmdutil.DryRunServer:
+		createOptions.DryRun = []string{metav1.DryRunAll}
```

Comment:
Looks like DryRunClient and DryRunServer does the same thing. If so, then why don't we combine both case into one to make it more clearer? Like
```
case cmdutil.DryRunClient, cmdutil.DryRunServer:
       createOptions.DryRun = []string{metav1.DryRunAll}
```

Reply:
good point! fixed.